site/viamguides/src/control-module/control-module.md

```diff
@@ -0,0 +1,577 @@
+   - **Language**: `Python`
+   - **Visibility**: `Private`
+   - **Namespace/Organization ID**: In the Viam app, navigate to your organization settings through the menu in upper right corner of the page. Find the Public namespace and copy that string.
+     ![public namespace](assets/namespace.png)
+   - **Resource to be added to the module**: `Generic Service`
+     - This is a [Generic Service](https://docs.viam.com/dev/reference/apis/services/generic/). If you choose a different resource type to add your control logic to, you must implement any required API methods for the chosen component.
+   - **Model name**: `lamp-alarm`
+     - This can be anything you choose to identify your control logic model.
+   - **Enable cloud build**: `Yes`
+     - This enables Viam to build your module via GitHub action.
+   - **Register module**: `Yes`
+     - This lets Viam know that the module exists.
+
+1. Press the **Enter** key and the generator will create a folder for your control logic component.
+
+   ![gif of running viam module generator and prompt selections](assets/generateModule-720p.gif)
+
+1. Open the code files using your preferred IDE. Find the Python file within the `src/models/` directory with the same name as your model name, `lamp_alarm.py` shown here. This is where we can add our control logic.
+   ![generated code files](assets/stub.png)
+
+Next, let's add our control logic.
+
+<!-- ------------------------ -->
+
+## Develop the module
+
+Duration: 5
+
+Now that you created the initial module, let's add our control logic, and deploy it to our Raspberry Pi.
+
+**Final code sample**: You can refer to [the sample code](https://github.com/loopDelicious/control-module-example/blob/main/src/models/lamp_alarm.py) to see how everything fits together. In this section, let's break it down step-by-step to learn how the module works.
+
+### Update the module code
+
+![generated code files](assets/stubLabeled.png)
+
+1.  **Review scaffolding**: Let's take a closer look at the generated files, and find the Python file within the `src/models/` directory with the same name as your model name, `lamp_alarm.py` shown here.
+    - Review the other generated files within the project directory
+    - Review the imports within `lamp_alarm.py`
+    - Review the class definition within `lamp_alarm.py`
+      - It includes the public namespace, module name, and model name that you specified in the module generation prompts (`joyce`, `control-lamp-alarm`, and `lamp-alarm` shown here).
+      - It also includes four default class methods `new()`, `validate_config()`, `reconfigure()`, and `do_command()`.
+1.  **Add dependencies**: Add these imports near the top of the file to support our control loop and provide more visibility during debugging. If your control logic requires external dependencies, be sure to also update the `requirements.txt` file.
+
+    ```python
+    import asyncio
+    from threading import Event
+    from viam.logging import getLogger
```

Comment:
We shouldn't need `from viam.logging import getLogger` anymore because there is a `self.logger` property included in the generated class that inherits from EasyResource. 
```suggestion
```